New Rats update breaks game

Love the update, however, looks like it isn't working

 

[11:16:03] [Render thread/ERROR]: There was a problem reading power file ratsmischief:ratified (skipping): Power type "requiem:soul_type" is not defined.
[11:16:03] [Render thread/ERROR]: Origin "ratsmischief:rat" contained unregistered power: "ratsmischief:ratified"

 

and because you added the requiem thing, I now cannot start a world due to it trying to load stuff from requiem, which I don't use.

 

 

 

[11:16:03] [Render thread/WARN]: Failed to load datapacks, can't proceed with server load

```java.util.concurrent.ExecutionException: io.github.apace100.calio.data.DataException: Error reading data field at condition: EntityCondition json type "requiem:start_remnant_gamerule" is not defined.
	at Not Enough Crashes deobfuscated stack trace.(1.17.1+build.61) ~[?:?]
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2069) ~[?:?]
	at net.minecraft.client.MinecraftClient.createIntegratedResourceManager(MinecraftClient:2153) ~[?:?]
	at net.minecraft.client.MinecraftClient.startIntegratedServer(MinecraftClient:1999) ~[?:?]
	at net.minecraft.client.MinecraftClient.createWorld(MinecraftClient:1943) ~[?:?]
	at net.minecraft.client.gui.screen.world.CreateWorldScreen.createLevel(CreateWorldScreen:279) ~[?:?]
	at net.minecraft.client.gui.screen.world.CreateWorldScreen.method_19922(CreateWorldScreen:215) ~[?:?]
	at net.minecraft.client.gui.widget.ButtonWidget.onPress(ButtonWidget:40) ~[?:?]
	at net.minecraft.client.gui.widget.PressableWidget.onClick(PressableWidget:16) ~[?:?]
	at net.minecraft.client.gui.widget.ClickableWidget.mouseClicked(ClickableWidget:121) ~[?:?]
	at net.minecraft.client.gui.ParentElement.mouseClicked(ParentElement:27) ~[?:?]
	at net.minecraft.client.Mouse.method_1611(Mouse:94) ~[?:?]
	at net.minecraft.client.gui.screen.Screen.wrapScreenError(Screen:490) ~[?:?]
	at net.minecraft.client.Mouse.onMouseButton(Mouse:94) ~[?:?]
	at net.minecraft.client.Mouse.method_22686(Mouse:165) ~[?:?]
	at net.minecraft.util.thread.ThreadExecutor.execute(ThreadExecutor:101) ~[?:?]
	at net.minecraft.client.Mouse.method_22684(Mouse:165) ~[?:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.2.2.jar:build 10]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3101) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem:172) ~[intermediary-1.17.1.jar:?]
	at net.minecraft.client.util.Window.swapBuffers(Window:310) ~[?:?]
	at net.minecraft.client.MinecraftClient.render(MinecraftClient:1137) ~[?:?]
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:728) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:217) [intermediary-1.17.1.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234) [fabric-loader-0.11.7.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153) [fabric-loader-0.11.7.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28) [fabric-loader-0.11.7.jar:?]
Caused by: io.github.apace100.calio.data.DataException: Error reading data field at condition: EntityCondition json type "requiem:start_remnant_gamerule" is not defined.
	at io.github.apace100.calio.data.SerializableData.lambda$read$2(SerializableData.java:91) ~[309eaadc-621b-4cbc-8501-83cede87f4fe.jar:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:723) ~[?:?]
	at io.github.apace100.calio.data.SerializableData.read(SerializableData.java:77) ~[309eaadc-621b-4cbc-8501-83cede87f4fe.jar:?]
	at io.github.apace100.origins.origin.OriginLayer$ConditionedOrigin.read(OriginLayer.java:328) ~[Origins-1.17-1.1.4.jar:?]
	at io.github.apace100.origins.origin.OriginLayer.lambda$merge$12(OriginLayer.java:133) ~[Origins-1.17-1.1.4.jar:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at io.github.apace100.origins.origin.OriginLayer.merge(OriginLayer.java:133) ~[Origins-1.17-1.1.4.jar:?]
	at io.github.apace100.origins.origin.OriginLayers.apply(OriginLayers.java:67) ~[Origins-1.17-1.1.4.jar:?]
	at io.github.apace100.origins.origin.OriginLayers.apply(OriginLayers.java:19) ~[Origins-1.17-1.1.4.jar:?]
	at net.minecraft.resource.SinglePreparationResourceReloader.method_18790(SinglePreparationResourceReloader:13) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
	at net.minecraft.resource.SimpleResourceReload.method_18365(SimpleResourceReload:71) ~[?:?]
	at net.minecraft.util.thread.ThreadExecutor.executeTask(ThreadExecutor:151) ~[?:?]
	at net.minecraft.util.thread.ReentrantThreadExecutor.executeTask(ReentrantThreadExecutor:23) ~[?:?]
	at net.minecraft.util.thread.ThreadExecutor.runTask(ThreadExecutor:125) ~[?:?]
	at net.minecraft.util.thread.ThreadExecutor.runTasks(ThreadExecutor:134) ~[?:?]
	at net.minecraft.client.MinecraftClient.createIntegratedResourceManager(MinecraftClient:2152) ~[?:?]
	... 29 more```

--- a/build.gradle
+++ b/build.gradle
@@ -49,6 +49,18 @@ repositories {
 		}
 	}
 	maven {
+		name "fall flying lib"
+		url "https://maven.cafeteria.dev"
+	}
+	maven {
+		name "cloth config"
+		url "https://maven.shedaniel.me/"
+	}
+	maven {
+		name "mod menu"
+		url "https://maven.terraformersmc.com/"
+	}
+	maven {
 		url = 'https://jitpack.io'
 		content {
 			includeGroupByRegex '(com|io)\\.github\\..*'
@@ -79,11 +91,13 @@ dependencies {
 
     modImplementation "io.github.ladysnake:requiem-api:${requiem_version}"
     modRuntime "io.github.ladysnake:requiem-core:${requiem_version}"
-    modRuntime "io.github.ladysnake:requiem:${requiem_version}"
+	modImplementation "io.github.ladysnake:requiem:${requiem_version}"
 
 	// Made available when Requiem is present
 	modCompileOnly "io.github.ladysnake:PlayerAbilityLib:${pal_version}"
 	modCompileOnly "io.github.onyxstudios.Cardinal-Components-API:cardinal-components-base:3.+"
+
+	modImplementation "com.github.apace100:origins-fabric:${origins_version}"
 
 	runtimeOnly "me.zeroeightsix:fiber:0.23.0-2"
 

--- a/gradle.properties
+++ b/gradle.properties
@@ -23,7 +23,7 @@ pal_version = 1.3.0
 findbugs_version = 3.0.2
 jb_annotations_version = 15.0
 apiguardian_version = 1.0.0
-origins_version = 1.0.2
+origins_version = 1.1.4
 
 #Publishing
 owners = Ladysnake

--- a/src/main/java/ladysnake/ratsmischief/common/Mischief.java
+++ b/src/main/java/ladysnake/ratsmischief/common/Mischief.java
@@ -3,6 +3,7 @@ package ladysnake.ratsmischief.common;
 import ladysnake.ratsmischief.common.armormaterials.RatMaskArmorMaterial;
 import ladysnake.ratsmischief.common.command.PlayerRatifyCommand;
 import ladysnake.ratsmischief.common.command.PlayerUnratifyCommand;
+import ladysnake.ratsmischief.common.compat.MischiefOriginsCompat;
 import ladysnake.ratsmischief.common.entity.RatEntity;
 import ladysnake.ratsmischief.common.item.RatPouchItem;
 import ladysnake.ratsmischief.common.item.RatStaffItem;
@@ -14,6 +15,7 @@ import net.fabricmc.fabric.api.event.lifecycle.v1.ServerTickEvents;
 import net.fabricmc.fabric.api.object.builder.v1.entity.FabricDefaultAttributeRegistry;
 import net.fabricmc.fabric.api.object.builder.v1.entity.FabricEntityTypeBuilder;
 import net.fabricmc.fabric.api.object.builder.v1.trade.TradeOfferHelper;
+import net.fabricmc.loader.api.FabricLoader;
 import net.minecraft.entity.*;
 import net.minecraft.entity.decoration.painting.PaintingMotive;
 import net.minecraft.item.ArmorItem;
@@ -29,8 +31,6 @@ import software.bernie.geckolib3.GeckoLib;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
-import java.util.Calendar;
-import java.util.Date;
 
 public class Mischief implements ModInitializer {
     public static final String MODID = "ratsmischief";
@@ -78,6 +78,11 @@ public class Mischief implements ModInitializer {
         CommandRegistrationCallback.EVENT.register((commandDispatcher, b) ->
                 PlayerUnratifyCommand.register(commandDispatcher)
         );
+        
+        // origins compat hack
+        if (FabricLoader.getInstance().isModLoaded("origins")) {
+            MischiefOriginsCompat.init();
+        }
 
         // rat custom spawner
         RatSpawner ratSpawner = new RatSpawner();

--- a/src/main/java/ladysnake/ratsmischief/common/compat/MischiefOriginsCompat.java
+++ b/src/main/java/ladysnake/ratsmischief/common/compat/MischiefOriginsCompat.java
@@ -1,0 +1,28 @@
+package ladysnake.ratsmischief.common.compat;
+
+import io.github.apace100.apoli.power.factory.condition.ConditionFactory;
+import io.github.apace100.apoli.registry.ApoliRegistries;
+import io.github.apace100.calio.data.SerializableData;
+import io.github.apace100.calio.data.SerializableDataTypes;
+import ladysnake.ratsmischief.common.Mischief;
+import ladysnake.requiem.common.gamerule.RequiemSyncedGamerules;
+import ladysnake.requiem.common.gamerule.StartingRemnantType;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+public class MischiefOriginsCompat {
+	public static final Identifier CONDITION_ID = new Identifier(Mischief.MODID, "enable_rat");
+	public static final ConditionFactory<Entity> CONDITION_FACTORY = new ConditionFactory<>(CONDITION_ID, new SerializableData().add("enabled", SerializableDataTypes.BOOLEAN), (instance, entity) -> {
+		if (FabricLoader.getInstance().isModLoaded("requiem")) {
+			StartingRemnantType startingRemnantType = RequiemSyncedGamerules.get(entity.world).getStartingRemnantType();
+			return startingRemnantType.getRemnantType() == StartingRemnantType.CHOOSE.getRemnantType();
+		}
+		return false;
+	});
+	
+	public static void init() {
+		Registry.register(ApoliRegistries.ENTITY_CONDITION, CONDITION_ID, CONDITION_FACTORY);
+	}
+}

--- a/src/main/resources/data/origins/origin_layers/origin.json
+++ b/src/main/resources/data/origins/origin_layers/origin.json
@@ -3,8 +3,8 @@
   "origins": [
     {
       "condition": {
-          "type": "requiem:start_remnant_gamerule",
-          "value": ["CHOOSE"]
+          "type": "ratsmischief:enable_rat",
+          "enabled": true
       },
       "origins": ["ratsmischief:rat"]
     }

